index.js

```diff
@@ -99,6 +99,23 @@ class Ora {
 	}
 }
 
+function promise(action, options) {
+	if (typeof action.then === 'function') {
+		const spinner = new Ora(options);
+		spinner.start();
+		action.then(() => {
+			spinner.succeed();
+		},
+		() => {
+			spinner.fail();
+		});
+	} else {
+		throw new Error('Parameter `action` must be a Promise');
+	}
+}
```

Comment:
Should return the `spinner` so it can be chained. That should also be documented like the other chainable methods are.